buildSrc/src/main/java/org/springframework/boot/build/bom/Library.java

```diff
@@ -541,7 +541,7 @@ public String toString() {
 
 	public record Link(String rootName, Function<LibraryVersion, String> factory, List<String> packages) {
 
-		private static final Pattern PACKAGE_EXPAND = Pattern.compile("^(.*)\\[(.*)\\]$");
+		private static final Pattern PACKAGE_EXPAND = Pattern.compile("^(.*)\\[(.*)]$");
```

Comment:
We're still escaping the opening `[` but the closing `]` is no longer escaped. I think the parser may cope with that but I find the expression harder to read as you have to know that the parser treats the `]` as a plain `]` rather than closing a never-opened character set. Please revert.